lib/active_shipping/carriers/ups.rb

```diff
@@ -684,6 +735,28 @@ def parse_tracking_response(response, options = {})
                            :tracking_number => tracking_number)
     end
 
+    def parse_delivery_dates_response(origin, destination, packages, response, options={})
+      xml     = build_document(response, 'TimeInTransitResponse')
+      success = response_success?(xml)
+      message = response_message(xml)
+      delivery_estimates = []
+
+      if success
+        xml.css('ServiceSummary').each do |service_summary|
+          # vvvvv Translate the Time in Transit Codes to the service codes used elsewhere vvvvv
```

Comment:
haha I actually just put that in to draw attention to that comment since I was pissed off that UPS decided to use two different sets of service codes in the API. But I'll take that out.